src/DependencyGraph.ts

```diff
@@ -137,17 +137,19 @@ export class Node {
     ) {
         if (dependencies.length > 0) {
             this.subscriptions = [];
-        }
-        for (let dependency of this.dependencies) {
-            let sub = this.graph.onchange(dependency, (event) => {
-                //notify the graph that we changed since one of our dependencies changed
-                this.graph.emit(this.key, event);
-            });
 
-            this.subscriptions.push(sub);
+            for (let dependency of this.dependencies) {
+                let sub = this.graph.onchange(dependency, (event) => {
+                    //notify the graph that we changed since one of our dependencies changed
+                    this.graph.emit(this.key, event);
+                });
+
+                this.subscriptions.push(sub);
+            }
```

Comment:
The change here is to only iterate through the dependencies if their length is > 0. This isn't a behavioral change, but moving the loop inside the conditional makes it easier to tell TS that `subscriptions` has the value `[]`. The difference between `subscriptions` being `[]` vs `undefined` is important in other places in this file, so I can't do something like always giving it a default value.